### PR TITLE
Allow ICMP to/from LXD networks

### LINKED ISSUE #9020: Allow IPv6 SLAAC/RA through LXD generated firewall rules
If the host's default INPUT and OUTPUT chains are set to DROP/REJECT, then LXD cannot send IPv6 router advertisements nor can the instances request such a response for SLAAC configuration.

LXD already works around this issue for IPv4 by allowing DHCP into the LXD bridge interface.

So we should add rules to allow similar for IPv6 auto configuration.

See https://discuss.linuxcontainers.org/t/no-ipv6-address/11576/5?u=tomp

### PR DESCRIPTION
**Pull request by stgraber**
Fixes #9020

### CODE REVIEW

File: lxd/firewall/drivers/drivers_consts.go [hunk 1]
```diff
@@ -10,8 +10,8 @@ const FilterIPv4All = "0.0.0.0"
 
 // FeatureOpts specify how firewall features are setup.
 type FeatureOpts struct {
-	DHCPDNSAccess   bool // Add rules to allow DHCP and DNS access.
-	ForwardingAllow bool // Add rules to allow IP forwarding. Blocked if false.
+	ICMPDHCPDNSAccess bool // Add rules to allow DHCP and DNS access.
```
Comment (tomponline) [inline lxd/firewall/drivers/drivers_consts.go:13]: Comment here could do with updating.

File: lxd/firewall/drivers/drivers_nftables.go [hunk 2]
```diff
@@ -261,8 +261,8 @@ func (d Nftables) networkSetupOutboundNAT(networkName string, SNATV4 *SNATOpts,
 	return nil
 }
 
-// networkSetupDHCPDNSAccess sets up basic nftables overrides for DHCP/DNS.
-func (d Nftables) networkSetupDHCPDNSAccess(networkName string, ipVersions []uint) error {
+// networkSetupICMPDHCPDNSAccess sets up basic nftables overrides for DHCP/DNS.
```
Comment (tomponline) [inline lxd/firewall/drivers/drivers_nftables.go:264]: comment update here too.

File: lxd/firewall/drivers/drivers_xtables.go [hunk 3]
```diff
@@ -382,6 +382,11 @@ func (d Xtables) networkSetupDHCPDNSAccess(networkName string, ipVersion uint) e
 			{"4", networkName, "filter", "OUTPUT", "-o", networkName, "-p", "udp", "--sport", "67", "-j", "ACCEPT"},
 			{"4", networkName, "filter", "OUTPUT", "-o", networkName, "-p", "udp", "--sport", "53", "-j", "ACCEPT"},
 			{"4", networkName, "filter", "OUTPUT", "-o", networkName, "-p", "tcp", "--sport", "53", "-j", "ACCEPT"}}
+
+		for _, icmpType := range []int{3, 11, 12} {
```
Comment (tomponline) [inline lxd/firewall/drivers/drivers_xtables.go:386]: Add comment " // Allow core ICMPv4 to/from LXD host."

File: lxd/firewall/drivers/drivers_xtables.go [hunk 4]
```diff
@@ -371,8 +371,8 @@ func (d Xtables) networkSetupOutboundNAT(networkName string, subnet *net.IPNet,
 	return nil
 }
 
-// networkSetupDHCPDNSAccess sets up basic iptables overrides for DHCP/DNS.
-func (d Xtables) networkSetupDHCPDNSAccess(networkName string, ipVersion uint) error {
+// networkSetupICMPDHCPDNSAccess sets up basic iptables overrides for DHCP/DNS.
```
Comment (tomponline) [inline lxd/firewall/drivers/drivers_xtables.go:374]: comment about icmp here too please.

**Review by tomponline — APPROVED** [review]
Looks good, just comments about comments :)

**Comment by stgraber** [comment]
@tomponline comments updated!

**Pull request merged**